algorithms/backtrack/generate_parenthesis.py

```diff
@@ -27,3 +27,17 @@ def add_pair(res, s, left, right):
     res = []
     add_pair(res, "", n, 0)
     return res
+
+
+def generate_parenthesis_(n):
```

Comment:
Please add tests in the tests folder ([tests/test_backtrack.py](https://github.com/keon/algorithms/blob/master/tests/test_backtrack.py)). 
Rename the new version as `generate_parenthesis_v2()`, also change the original version name to `generate_parenthesis_v1()`.